service/s3/internal/customizations/updateEndpoint.go

```diff
@@ -0,0 +1,156 @@
+package customizations
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"regexp"
+	"strings"
+
+	"github.com/awslabs/smithy-go"
+	"github.com/awslabs/smithy-go/middleware"
+	"github.com/awslabs/smithy-go/transport/http"
+)
+
+// UpdateEndpointOptions provides the options for the UpdateEndpoint middleware setup.
+type UpdateEndpointOptions struct {
```

Comment:
Do we need this type? Can we just pass the two boolean value directly to the constructor function?

Reply:
We want to keep the struct here. The struct helps with adding function pointers and possible options without having to change the constructor functions. Also we want to stay consistent with other customizations. This is all internal, so can be modified later if needed